### PR TITLE
Make sure exif is added if exists and not addet if null or 0.0

### LINKED ISSUE #3684: {{Location|0.0|0.0}}
**Summary:** 

The app adds _{{Location|0.0|0.0}}_ to the file description although the files metadata does not contain any GPS data. Example: [File:Renaturierter Tauberabschnitt zwischen Tauberbischofsheim und Impfingen 8.jpg](https://commons.wikimedia.org/wiki/File:Renaturierter_Tauberabschnitt_zwischen_Tauberbischofsheim_und_Impfingen_8.jpg)

This is wrong and, thus, the location should not appear in the wikitext.

Generally, (0.0, 0.0) is a strong indicator that something is wrong with the data.

### PR DESCRIPTION
**Pull request by neslihanturan**
**Description (required)**

Fixes #3684 and #4016 by making sure adding the coordinate if exists and avoiding adding it if 0.0, 0.0.

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/upload/ImageCoordinates.kt [hunk 1]
```diff
@@ -34,20 +34,28 @@ class ImageCoordinates internal constructor(exif: ExifInterface?) {
...
-                decLatitude =
-                    if (ExifInterface.LATITUDE_NORTH == latitudeRef) convertToDegree(latitude)
-                    else 0 - convertToDegree(latitude)
-                decLongitude =
-                    if (ExifInterface.LONGITUDE_EAST == longitudeRef) convertToDegree(longitude)
-                    else 0 - convertToDegree(longitude)
+            val latAndLong = exif.latLong
+            if (latAndLong != null) {
+                decLatitude = latAndLong[0]
+                decLongitude = latAndLong[1]
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/upload/ImageCoordinates.kt:40]: Should we set `imageCoordsExists = true` here? (or move the one on line 48 to line 60)

**Review by domdomegg — APPROVED** [review]

File: app/src/main/java/fr/free/nrw/commons/upload/ImageCoordinates.kt (hunk 1 above)
Comment (neslihanturan) [reply app/src/main/java/fr/free/nrw/commons/upload/ImageCoordinates.kt:40]: Agreed, thanks a lot @domdomegg !

**Review by domdomegg — APPROVED** [review]
LGTM :)

**Comment by misaochan** [comment]
Thanks @neslihanturan !

**Pull request merged**